### PR TITLE
dev server hints if `paths.base` prefix is missing; `--open` uses `paths.base`

### LINKED ISSUE #4418: Redirect to base by default
### Describe the problem

When `kit.paths.base` is defined, by default the app opens to a "Not found" root route. The developer has to manually change the route to get to the desired location.

### Describe the proposed solution

It would be helpful if dev server redirected to base route by default.

### Alternatives considered

To solve the issue I created a vite plugin - [vite-plugin-redirect](https://github.com/ebeloded/vite-plugin-redirect), which works like this:
```js
import redirect from 'vite-plugin-redirect';

const base = '/admin';

/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    paths: {
      base
    },
    vite: {
      plugins: [redirect({ '/': base })]
    }
  }
};
```

Unless there is a reason to not redirect to base, I think this functionality should be built-in.

### Importance

would make my life easier

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Rich-Harris**
closes #4418 — if `path.base` is `/blah` and the developer visits `/foo` in dev, they are redirected to `/blah/foo`

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by Conduitry** [comment]
Is this a good idea? It feels like it's asking for trouble, although I'm admittedly not clear on what that trouble would look like exactly.

In https://github.com/sveltejs/kit/issues/4418#issuecomment-1075481738 I suggest another way of handling this.

**Comment by mrkishi** [comment]
Yeah, it does sound like arbitrary `/foo` redirects could mask broken links and fetches that wouldn't work in production.

**Comment by Rich-Harris** [comment]
Masking broken links is a good reason not to do it. Updated the PR

**Comment by Conduitry** [comment]
:+1: The changeset message and the PR title should be updated as well.

File: .changeset/lovely-hairs-applaud.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+Add missing paths.base prefix in dev, if necessary
```
Comment (Rich-Harris) [inline .changeset/lovely-hairs-applaud.md:5]:
```suggestion
Hint if `paths.base` is missing in dev
```

File: .changeset/new-rules-vanish.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@sveltejs/kit": patch
```
Comment (Conduitry) [inline .changeset/new-rules-vanish.md:2]: Is it going to matter if these are double quotes instead of single quotes?

Comment (Rich-Harris) [reply .changeset/new-rules-vanish.md:2]: this was created via the link in the comment — https://github.com/sveltejs/kit/pull/4510#issuecomment-1087947182 — so i certainly hope not

Comment (Conduitry) [reply .changeset/new-rules-vanish.md:2]: Huh, weird that the CLI does something different than the bot. Anyway, if this wasn't the result of you manually writing the file, I'm not worried.

**Review by Conduitry — APPROVED** [review]

File: packages/kit/src/core/dev/plugin.js [hunk 3]
```diff
@@ -203,7 +204,13 @@ export async function create_plugin(config, cwd) {
 
 						if (req.url === '/favicon.ico') return not_found(res);
 
-						if (!decoded.startsWith(config.kit.paths.base)) return not_found(res);
+						if (!decoded.startsWith(config.kit.paths.base)) {
+							const suggestion = normalize_path(
+								config.kit.paths.base + req.url,
+								config.kit.trailingSlash
+							);
+							return not_found(res, `Not found (did you mean ${suggestion}?)`);
```
Comment (mrkishi) [inline packages/kit/src/core/dev/plugin.js:212]: Perhaps?
```suggestion
							return not_found(res, `Not found (did you mean <a href="${suggestion}">${suggestion}</a>?)`);
```

Comment (Conduitry) [reply packages/kit/src/core/dev/plugin.js:212]: This error message gets returned in a plain text response (with no `content-type`). I don't think it's worth it to make it an HTML response and then worry about escaping the suggestion appropriately.

Comment (Rich-Harris) [reply packages/kit/src/core/dev/plugin.js:212]: yeah, that's the journey i went on too

**Pull request merged**